middlewares/user/getTargetUserIdByPMChannelId.js

```diff
@@ -0,0 +1,57 @@
+const {StreamChat} = require('stream-chat');
+const Getstream = require('../../vendor/getstream');
+
+/**
+ *
+ * @param {import('express').Request} req
+ * @param {import('express').Response} res
+ * @param {import('express').NextFunction} next
+ * @returns
+ */
+const getTargetUserIdByPMChannelIdMiddleware = async (req, res, next) => {
+  const {userId} = req || {};
+  const {channelId} = req?.query || {};
+
+  if (!channelId) {
+    return res.status(400).json({
+      code: 400,
+      success: false,
+      message: 'Channel ID is required'
+    });
+  }
+
+  const chatClient = new StreamChat(process.env.API_KEY, process.env.SECRET);
+  let channelMembers;
+  try {
+    let {users} = await Getstream.chat.getTargetUserIdByChannelId(chatClient, userId, channelId);
+    channelMembers = users;
+  } catch (e) {
+    console.error('Failed to get channel members', e);
+    return res.status(500).json({
+      code: 500,
+      success: false,
+      message: 'Internal server error'
+    });
+  }
+
+  if (!channelMembers || channelMembers?.length !== 1)
+    return res.status(400).json({
+      code: 400,
+      success: false,
+      message: 'Not a valid PM channel'
+    });
+
+  const targetUserId = channelMembers?.at(0)?.user_id;
+
+  if (!targetUserId)
+    return res.status(400).json({
+      code: 400,
+      success: false,
+      message: "The user on this channel doesn't have a valid user id"
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code assumes that `channelMembers.at(0)` will always return an object and tries to access `user_id` property on it. This could lead to a TypeError if `channelMembers.at(0)` is undefined or not an object.
```diff
-  const targetUserId = channelMembers?.at(0)?.user_id;
+  const firstMember = Array.isArray(channelMembers) ? channelMembers[0] : undefined;
+  const targetUserId = firstMember ? firstMember.user_id : undefined;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->